core/utils.js

```diff
@@ -402,29 +403,38 @@ Blockly.utils.tokenizeInterpolation_ = function(message,
 };
 
 /**
- * Generate a unique ID.  This should be globally unique.
- * 87 characters ^ 20 length > 128 bits (better than a UUID).
+ * Generate a unique ID.
+ *
+ * This is actually a get/set accessor pair for
+ * Blockly.utils.idGenerator.genUid so that existing code that stubs
+ * genUid (including code in
+ * blockly-samples/plugins/dev-tools/src/block_test_helpers.mocha.js)
+ * can continue to stub/mock this function until we provide a better
+ * mechansim for doing so.
+ *
  * @return {string} A globally unique ID string.
+ * @deprecated Use Blockly.utils.idGenerator.genUid instead.
  */
-Blockly.utils.genUid = function() {
-  var length = 20;
-  var soupLength = Blockly.utils.genUid.soup_.length;
-  var id = [];
-  for (var i = 0; i < length; i++) {
-    id[i] = Blockly.utils.genUid.soup_.charAt(Math.random() * soupLength);
+Blockly.utils.genUid = function() {};  // For type declaration.
+
+Object.defineProperties(Blockly.utils, {
```

Comment:
We no longer need the getter/setter pair. `genUid` can just do a `deprecation.warn` and then call `utils.idGenerator.genUid`.

Reply:
Fixed.